dynatrace-oneagent/deploy-dynatrace-oneagent.sh

```diff
@@ -16,6 +29,17 @@ if [ -z "$DT_PAAS_TOKEN" ]; then
 	exit 1
 fi
 
+function apply_platform_specifics_to_yaml_file() {
+
+    PLATFORM=$1; FILE=$2
+    if [ "$PLATFORM" = "KUBERNETES" ]; then
+      echo "" >> $FILE
+      echo "  env:" >> $FILE
+      echo "    - name: ONEAGENT_ENABLE_VOLUME_STORAGE" >> $FILE
+      echo "      value: \"true\"" >> $FILE
```

Comment:
Unfortunately you ran into a well-known issue of OneAgent on GKE. You probably did not change the (by default) selected container-optimized OS, as documented in our docs: https://keptn.sh/docs/0.7.x/monitoring/dynatrace/install/#notes

In our [tutorial (Step 3)](https://tutorials.keptn.sh/tutorials/keptn-installation-gke-07/index.html?index=..%2F..index#2), we specifically ask users to select `ubuntu` when creating a new cluster for GKE.

Nevertheless, we should make sure our script works out of the box. I guess it would be best to check which cluster we are deploying to?

Reply:
After looking into this issue, we can apply the `ONEAGENT_ENABLE_VOLUME_STORAGE=true` fix if
```
kubectl get nodes -l 'cloud.google.com/gke-os-distribution=cos'
```
returns any nodes.